tests/Oryx.Integration.Tests/DotNetCore/DotNetCorePreRunCommandOrScriptTest.cs

```diff
@@ -0,0 +1,137 @@
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .SetEnvironmentVariable(FilePaths.PreRunCommandOrScriptEnvVarName, "echo \"Here has a pre run command.\"")
+                .AddCommand(
+                $"oryx create-script -appPath {appOutputDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .ToString();
+
+            await EndToEndTestHelper.BuildRunAndAssertAppAsync(
+                NetCoreApp30WebApp,
+                _output,
+                new[] { volume },
+                _imageHelper.GetGitHubActionsBuildImage(),
+                "/bin/sh",
+                new[]
+                {
+                    "-c",
+                    buildImageScript
+                },
+                _imageHelper.GetTestRuntimeImage("dotnetcore", "prerun"),
+                ContainerPort,
+                "/bin/sh",
+                new[]
+                {
+                    "-c",
+                    runtimeImageScript
+                },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
+                    Assert.Contains(expectedResponseContent, data);
+                });
+        }
+
+        [Theory]
+        [InlineData("2.1", NetCoreApp21WebApp, "Hello World!")]
+        [InlineData("3.1", NetCoreApp31MvcApp, "Welcome to ASP.NET Core MVC!")]
+        public async Task CanBuildAndRun_NetCore31WebApp_UsingPreRunScript_WithDynamicInstalls(
+            string runtimeVersion,
+            string appName,
+            string expectedResponseContent)
+        {
+            // Arrange
+            var hostDir = Path.Combine(_hostSamplesDir, "DotNetCore", appName);
+            var volume = DockerVolume.CreateMirror(hostDir);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = $"{appDir}/myoutputdir";
+            var buildImageScript = new ShellScriptBuilder()
+               .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+               .AddCommand(
+                $"oryx build {appDir} --platform dotnet --platform-version {runtimeVersion} -o {appOutputDir}")
+               .ToString();
+            var runtimeImageScript = new ShellScriptBuilder()
+                .SetEnvironmentVariable(SettingsKeys.EnableDynamicInstall, true.ToString())
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .SetEnvironmentVariable(FilePaths.PreRunCommandOrScriptEnvVarName, "./prerunscript.sh")
+                .AddCommand(
```

Comment:
Wondering how we're going to test for prerunscript.sh file on this integration test? Since there doesn't exist one, where should we create one for testing?

Reply:
Just before this line you can add a command which creates this file called `prerunscript.sh` in the `appOutputDir`. Also make sure you put some content in the `prerunscript.sh` in such a way that you can Assert that indeed this script ran